libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/columnChart/PluggableColumnChart.tsx

```diff
@@ -76,47 +77,25 @@ export class PluggableColumnChart extends PluggableColumnBarCharts {
                 return {
                     defaultSort,
                     availableSorts: [
-                        {
-                            itemId: localIdRef(viewBy[0].localIdentifier),
-                            attributeSort: {
-                                normalSortEnabled: true,
-                                areaSortEnabled: true,
-                            },
-                        },
-                        {
-                            itemId: localIdRef(viewBy[1].localIdentifier),
-                            attributeSort: {
-                                normalSortEnabled: true,
-                                areaSortEnabled: true,
-                            },
-                            metricSorts: [
-                                ...measures.map((m) => newMeasureSortSuggestion(m.localIdentifier)),
-                            ],
-                        },
+                        newAvailableSortsGroup(viewBy[0].localIdentifier),
+                        newAvailableSortsGroup(
+                            viewBy[1].localIdentifier,
+                            measures.map((m) => m.localIdentifier),
+                        ),
                     ],
                 };
             }
 
             return {
                 defaultSort,
                 availableSorts: [
-                    {
-                        itemId: localIdRef(viewBy[0].localIdentifier),
-                        attributeSort: {
-                            normalSortEnabled: true,
-                            areaSortEnabled: true,
-                        },
-                    },
-                    {
-                        itemId: localIdRef(viewBy[1].localIdentifier),
-                        attributeSort: {
-                            normalSortEnabled: true,
-                            areaSortEnabled: isStacked || measures.length > 1,
-                        },
-                        metricSorts: isEmpty(stackBy)
-                            ? [...measures.map((m) => newMeasureSortSuggestion(m.localIdentifier))]
-                            : [],
-                    },
+                    newAvailableSortsGroup(viewBy[0].localIdentifier),
+                    newAvailableSortsGroup(
+                        viewBy[1].localIdentifier,
+                        isEmpty(stackBy) ? [...measures.map((m) => m.localIdentifier)] : [],
```

Comment:
Why is there array copy here? It is not necessary, `map` already does a new array.

Reply:
fixed on all places